src/User/SignupService.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\User;
+
+final class SignupService
+{
+    private UserLogin $login;
+    private UserPassword $password;
```

Comment:
That looks wrong since we are having state here and services in general should not have state. These two look like they should be part of the singup() method i.e. `signup($login, $password)`.